dislib/data/classes.py

```diff
@@ -101,6 +135,19 @@ def subset_size(self, index):
 
         return self._sizes[index]
 
+    def subsets_sizes(self):
+        """ Returns the number of samples in all the Subsets.
+        If the size is unknown, this method performs a synchronization on
+        Subset.samples.shape[0] for all subsets.
+
+        Returns
+        -------
+        subsets_sizes : ndarray
+            Number of samples in each subset.
+        """
+
+        return [self.subset_size(i) for i in range(len(self))]
```

Comment:
Maybe it would be better to get all the sizes first (call _subset_size multiple times), and the synchronize on a list of futures instead of synchronizing multiple times on every size.

Reply:
Ok. I'll change it. But I didn't want to duplicate the code of the other function for what I think it'll be a minimal speed up.